src/OSPSuite.SimModel/include/SimModel/BooleanFormula.h

```diff
@@ -40,8 +40,6 @@ class BooleanFormula :
 		void BooleanFormula::SwitchFormulaFromComparisonFormula(std::vector<Formula*> &vecExplicit, std::vector<Formula*> &vecImplicit);
 
 		virtual void UpdateIndicesOfReferencedVariables();
-		//Update the value to ODEScaleFactor of the scale factor of the variable with the id referenced by this formula.
-		virtual void UpdateScaleFactorOfReferencedVariable(const int quantity_id, const double ODEScaleFactor);
```

Comment:
The original implementation of scale factor update was too complex.
In the original implementation the "top-level" Formulas using a species were cached, after that the 
`UpdateScaleFactorOfReferencedVariable`-call was propagated through the whole formula tree.
Effectively used was it only in the "low level" formulas (VariableFormula, SimpleProductFormula.

In the new implementation, the low level formulas are cahced directly in the using species, so that there is no need to define `UpdateScaleFactorOfReferencedVariable `everywhere